opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -180,6 +180,26 @@ where
         // Visit fields.
         event.record(&mut visitor);
 
+        #[cfg(feature = "experimental_use_tracing_span_context")]
+        if let Some(span) = _ctx.event_span(event) {
+            use tracing_opentelemetry::OtelData;
+            let opt_span_id = span
+                .extensions()
+                .get::<OtelData>()
+                .and_then(|otd| otd.builder.span_id);
+
+            let opt_trace_id = span.scope().last().and_then(|root_span| {
```

Comment:
Is it really necessary to look for the _root_ span to get the `trace_id` here? Shouldn't the current span have the `trace_id` set as well?

Reply:
Typically the current span won't have the span id explicitly set on it, so it won't actually store the trace id itself, instead it will be inherited when the span is built for sending.

Reply:
I suppose going straight to the root span wouldn't support an explicitly set trace id on a non-root span.  Is there a non-contrived example of when this might be done that I should add a test for and support?  Or should this just be left as a limitation to avoid extra code and runtime overhead?

Reply:
I looked at the listener code and yeah, it's only _root_ spans that get the `trace_id` set in the builder. Someone would have to manually override that and set a `trace_id` on the builder to get that behavior.

Still, I think that we should pull out the `trace_id` from the builder and use that if it's available, or fall back to the _root_ one.